pkg/cfn/template/iam.go

```diff
@@ -25,7 +25,7 @@ type IAMRole struct {
 	Path string `json:",omitempty"`
 
 	AssumeRolePolicyDocument MapOfInterfaces `json:",omitempty"`
-	ManagedPolicyArns        []string        `json:",omitempty"`
+	ManagedPolicyArns        []interface{}   `json:",omitempty"`
```

Comment:
I'm not clear why this is changed. Is it just because we're smuggling the return value from `makePolicyARN` out of `(rs *IAMServiceAccountResourceSet) AddAllResources()` inside this slice? Would it make sense to just stringify the value before adding it to `ManagedPolicyArns` instead?

(I'm not clear on what this `gfn` business is all about, so maybe this comment doesn't make sense.)

Reply:
Yes it's kind of a less than ideal situation because we do some cloud formation using our own types and some using goformation. It's here that they're being mixed. `makePolicyARN` uses Cloudformation intrinsics, which aren't serialized as strings. Although in this case it's probably not strictly necessary to use the intrinsic.

Reply:
I'm going to leave the intrinsics because we use them for this all through out cloudformation

Reply:
I'm going to leave the intrinsics because we use them elsewhere (with `makePolicyArns`) (and `interface{}` is sort of how things are done in `cfn/template` right now)